Sources_v3/StreamChatUI/Common Views/OnlyLinkTappableTextView.swift

```diff
@@ -0,0 +1,19 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+/// Text View that ignore all user interactions except touches on links
+class OnlyLinkTappableTextView: UITextView {
+    override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
+        guard
+            let range = characterRange(at: point),
+            !range.isEmpty,
+            let position = closestPosition(to: point, within: range),
+            let styles = textStyling(at: position, in: .forward),
+            styles[.link] != nil
+        else { return nil }
+        return self
```

Comment:
What's wrong with `if { } else { }`? :)

Reply:
Also, we should probably call `super` here, not just return `self`

Reply:
> What's wrong with if { } else { }? :)
All 3 calls return optional. So I either need to chain it with `Optional.flatMap`, use `if let` or `guard`.
Later is question of preferences.

Reply:
> What do you mean exactly by `Missing a lot of system features.` in 4) ?

Drag&drop of links, link preview, probably something else.

Reply:
I'd argue that it's a question of preference. 
`guard` means - it this condition is not met, the rest of the method doesn't matter
`if {} else {}` means - there are just two options here, either this or that

This case is the latter imo.